### PR TITLE
fix(mgmt,billing): refactor mgmt message, endpoints and add billing service

### PR DESCRIPTION
**Pull request by xiaofei-du**
Because

- list, create, update, delete and lookup user operations should be admin API that is not exposed to the public internet

This commit

- add billing service so each user is assigned with a billing plan
- redesign mgmt-backend endpoints into Admin and Public APIs
- update the message user fields

### CODE REVIEW

File: vdp/billing/v1alpha/plan.proto [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+  };
+
+  // Tier enumerates the billing plan tier type
+  enum Tier {
+    // PlanType: UNSPECIFIED, equivalent to FREE.
+    TIER_UNSPECIFIED = 0;
+    // PlanType: FREE
+    TIER_FREE = 1;
+    // PlanType: CUSTOM
+    TIER_CUSTOM = 5;
```
Comment (pinglin) [inline vdp/billing/v1alpha/plan.proto:29]:
```suggestion
    TIER_CUSTOM = 2;
```

File: vdp/billing/v1alpha/plan.proto [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+// ListPlanRequest represents a request to list all plans
+message ListPlanRequest {
+  // Page size: the maximum number of resources to return. The service may
+  // return fewer than this value. If unspecified, at most 10 plans will be
+  // returned. The maximum value is 100; values above 100 will be coereced to
+  // 100.
+  optional int64 page_size = 1 [ (google.api.field_behavior) = OPTIONAL ];
+  // Page token
+  optional string page_token = 2 [ (google.api.field_behavior) = OPTIONAL ];
+}
```
Comment (pinglin) [inline vdp/billing/v1alpha/plan.proto:75]:
```suggestion
message ListPlanRequest {
  // Page size: the maximum number of resources to return. The service may
  // return fewer than this value. If unspecified, at most 10 plans will be
  // returned. The maximum value is 100; values above 100 will be coereced to
  // 100.
  optional int64 page_size = 1 [ (google.api.field_behavior) = OPTIONAL ];
  // Page token
  optional string page_token = 2 [ (google.api.field_behavior) = OPTIONAL ];
    // View view (default is VIEW_BASIC)
  optional View view = 3 [ (google.api.field_behavior) = OPTIONAL ];
  // Filter expression to list pipelines
  optional string filter = 4 [ (google.api.field_behavior) = OPTIONAL ];
}
```

**Review by pinglin — CHANGES_REQUESTED** [review]

File: vdp/billing/v1alpha/plan.proto (hunk 2 above)
Comment (xiaofei-du) [reply vdp/billing/v1alpha/plan.proto]: Please see the latest commit

**Review by pinglin — CHANGES_REQUESTED** [review]
Please update openapi.

**Comment by xiaofei-du** [comment]
> Please update openapi.

Please check the latest commit.

**Review by pinglin — APPROVED** [review]

**Pull request merged**